tests/phpunit/CRM/MembershipExtras/Hook/ValidateForm/MembesPaymentPlanTest.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+use CRM_MembershipExtras_Test_Fabricator_MembershipType as MembershipTypeFabricator;
+use CRM_MembershipExtras_Test_Fabricator_PriceSet as PriceSetFabricator;
+use CRM_MembershipExtras_Test_Fabricator_PriceField as PriceFieldFabricator;
+use CRM_MembershipExtras_Test_Fabricator_PriceFieldValue as PriceFieldValueFabricator;
+use CRM_MembershipExtras_Exception_InvalidMembershipTypeInstalment as InvalidMembershipTypeInstalment;
+
+/**
+ * Class CRM_MembershipExtras_Hook_ValidateForm_MembershipPaymentPlanTest
+ *
+ * @group headless
+ */
+class CRM_MembershipExtras_Hook_ValidateForm_MembershipPaymentPlanTest extends BaseHeadlessTest {
+
+  private $form;
+  private $errors;
+
+  public function setUp() {
+    $this->form = new CRM_Member_Form();
+    $this->errors = ['price_set_id' => NULL];
+  }
+
+  /**
+   * Tests error when mixed period membership type price field values are submitted.
+   */
+  public function testErrorWhenMixedPeriodMembershipTypePriceFieldValuesAreSubmitted() {
+    $priceSet = $this->mockPriceSet();
+    $priceField = $this->mockPriceField($priceSet['id'], 'Test Field Set 1');
+    $memType1 = $this->mockMembershipType('rolling', 'year');
+    $memType2 = $this->mockMembershipType('fixed', 'year');
+    $priceFieldValue1 = $this->mockPriceFieldValue($priceField['id'], $memType1['id']);
+    $priceFieldValue2 = $this->mockPriceFieldValue($priceField['id'], $memType2['id']);
+
+    $fields = [];
+    $fields['price_set_id'] = $priceSet['id'];
+    $mockedPriceFielddKey = 'price_' . (string) $priceField['id'];
```

Comment:
Shoud be `mockedPriceFieldKey` instead of `mockedPriceFielddKey`.

Reply:
@ahed-compucorp  good eyes!